EducationalAssignments/UndoPartTwo.md

```diff
@@ -0,0 +1,192 @@
+circumvention. Attackers must not be able to bypass, disable, or exploit the
+defense monitor's enhanced behaviors, ensuring the integrity and intended
+functionality of file operations.
+
+You will submit a zip file containing all of the tests you have created. You
+will gain points for every student's reference monitor you find a flaw in. It is
+good if multiple tests of yours break a student's reference monitor, but you
+gain the same number of tests whether one or more tests break the layer.
+
+
+## Prerequisites
+
+This assignment assumes you have both the latest Python 2.7 and RepyV2 installed
+on your computer. Please refer to the [SeattleTestbed Build
+Instructions](../Contributing/BuildInstructions.md#prerequisites) for
+information on how to get them.
+
+
+### Helpful links
+
+The following links will aid students in becoming comfortable with Python, Repy
+and seattle:
+ * Official [Python tutorial](http://docs.python.org/tutorial/)
+ * [Differences between RepyV2 and Python](../Programming/PythonVsRepyV2.md)
+ * List of [RepyV2 API calls](../Programming/RepyV2API.md)
+
+
+## Testing security layers
+
+### Hypothesis, test case, counter example
+
+The goal of a good tester is to test hypotheses.  A hypothesis is just a
+scientific way of asking a question.  The hypothesis of this assignment is "This
+security layer is well designed."  The questions you will ask when running your
+test cases will always be the same
+
+ * "Is this reference monitor secure?"
+
+ * "Does this reference monitor hamper performance?"
+
+ * "Does this reference monitor prevent actions that should be allowed?"
+
+Notice that these questions are parallels of the security paradigms: security,
+efficiency and accuracy, respectively.  
+
+If we can find a case where the hypothesis is false, then the security layer is
+not secure.  Such a case is referred to as a counter example.  Hence all test
+cases should be designed to test for these three types of flaws.
+
+#### Information on: Try, Except, Else, Finally
+
+The try, except, else and finally statements are part of **exception handling**.
+For more information on exception handling please visit:
+
+ * [http://docs.python.org/tutorial/errors.html]
+ * [http://wiki.python.org/moin/HandlingExceptions]
+ * [http://www.tutorialspoint.com/python/python_exceptions.htm]
+
+### Hints and Ideas for testing
+
+When writing your own tests it is important to test for a complete set of
+possible penetrations.  Keep in mind, it only takes one test case to break
+through a security layer.  Some of the things you may want to test for include:
+
+ * threading
+ * multiple writes 
+
+And more!  Remember a good security layer can't be broken by anyone!  Which is
+all part of the fun!  It's about solving a puzzle.  First you make the puzzle -
+write the security layer, then you solve the puzzle - try to bypass it.  If your
+puzzle is "good enough", no one will be able to break it, no matter what.  
+
```

Comment:
Please add some text urging them to make different test files for different items they want to test.  If they make an error in a test file, we will throw the whole test file out.

Reply:
I added a note
https://github.com/SeattleTestbed/docs/pull/84/files#diff-8e745d39f473445486d7a903fe30db6442684bf85a763ed6a2463bba52479cfaR122-R127

I added this in the assignment sheet as well.